Framework/DataHandling/src/LoadAscii.cpp

```diff
@@ -199,11 +199,15 @@ API::Workspace_sptr LoadAscii::readData(std::ifstream &file) const {
     }
 
     for (size_t i = 0; i < numSpectra; ++i) {
-      spectra[i].dataX().push_back(values[0]);
-      spectra[i].dataY().push_back(values[i * 2 + 1]);
+      auto resizedHistogram = spectra[i].histogram();
+      const size_t newPosition = resizedHistogram.x().size();
```

Comment:
Careful here. The size of a `Histogram` (and what `resize` takes is the number of *bins* (or values). So if you take the X length your resize below is not what you want in the case of X as bin-edges.

I think Anton moved `Histogram::size()` to `public`, so once his PR is merged you can use that here, that would be safer an cleaner, i.e., you can then just do `hist.resize(hist.size() + 1)`.